dascore/utils/patch.py

```diff
@@ -446,21 +431,90 @@ def get_start_stop_step(patch: PatchType, dim):
     return start, stop, step
 
 
-def get_default_patch_name(patch):
-    """Generates the name of the node."""
+def get_patch_names(
+    patch_data: pd.DataFrame | dc.Patch | dc.BaseSpool,
+    prefix="DAS",
+    attrs=("network", "station", "tag"),
+    coords=("time",),
+    sep="__",
+) -> pd.Series:
+    """
+    Generates the default name of patch data.
+
+    Parameters
+    ----------
+    prefix
+        A string to prefix the names.
+    patch_data
+        A container with patch data.
+    coords
+        The coordinate ranges to use for names.
+    sep
+        The separator for the strings.
+
+    Notes
+    -----
+    There are two special cases where the default logic is overwritten.
+    The first one, is when a column called "name" already exists. This
+    will simply be returned.
+
+    The second is when a column called "path" exists. In this case, the
+    output will be the file name with the extension removed. The path must
+    use / as a delinater.
 
-    def _format_datetime64(dt):
+    Examples
+    --------
+    >>> import dascore as dc
+    >>> from dascore.utils.patch import get_patch_names
+    >>> patch = dc.get_example_patch()
+    >>> name = get_patch_names(patch)
```

Comment:
`name` is:

```python
0    DAS_______random__2017_09_18__2017_09_18T00_00_07
Name: network, dtype: object
```

We get `DAS_______random__2017_09_18__2017_09_18T00_00_07` using `name[0]` because the function returns a `<class 'pandas.core.series.Series'>`. Is there a specific reason that it returns a pandas series instead of the name in a string?

Reply:
Yes, for the function to work with a spool it needs to return multiple names. I avoided returning multiple types (based on if the input is a Patch or not) from `get_patch_names` as that is a bit of an anti-pattern. 

I understand the confusion though. I will add `get_patch_name` as a method of `Patch` that will do this. I think with the different name (`get_patch_name` vs `get_patch_names`) it will be clear.